src/google/protobuf/descriptor.proto

```diff
@@ -481,7 +490,11 @@ message FieldOptions {
   // For Google-internal migration only. Do not use.
   optional bool weak = 10 [default=false];
 
-
+  // Provides the ability to override the name of the property
+  // generated for this field. This is applied to all properties
+  // and methods to do with this field, including HasFoo, FooCount,
+  // FooList etc.
+  optional string csharp_property_name = 11;
```

Comment:
We are actually adding a suffix "_" if we detect that containing type's name equal to name of the autogenerated property name (csharp_helpers.cc:264). so the property in our case would be called Name_, which would work but it's very ugly and it is reasonable to expect that users would want do override the name - which is what this option is for.

That said, if property_name option bothers you, we can remove it for now and it should be easy to reintroduce it later. I think we are at the point when we really need to move forward with this. Please advise.